functions/helloworld/test/sample.integration.pubsub.test.js

```diff
@@ -14,76 +14,100 @@
 
 // [START functions_pubsub_integration_test]
 const assert = require('assert');
-const execPromise = require('child-process-promise').exec;
-const path = require('path');
-const requestRetry = require('requestretry');
+const {spawn} = require('child_process');
+const {request} = require('gaxios');
 const uuid = require('uuid');
-
-const cwd = path.join(__dirname, '..');
+const waitPort = require('wait-port');
 
 // [END functions_pubsub_integration_test]
 
 describe('functions_helloworld_pubsub integration test', () => {
+  let ffProc;
+  afterEach(() => {
+    if (ffProc) {
+      ffProc.kill();
+    }
+  });
   // [START functions_pubsub_integration_test]
   it('helloPubSub: should print a name', async () => {
     const name = uuid.v4();
     const PORT = 8088; // Each running framework instance needs a unique port
 
     const encodedName = Buffer.from(name).toString('base64');
     const pubsubMessage = {data: {data: encodedName}};
-
-    // exec's 'timeout' param won't kill children of "shim" /bin/sh process
-    // Workaround: include "& sleep <TIMEOUT>; kill $!" in executed command
-    const proc = execPromise(
-      `functions-framework --target=helloPubSub --signature-type=event --port=${PORT} & sleep 1; kill $!`,
-      {shell: true, cwd}
-    );
+    ffProc = spawn('npx', [
+      'functions-framework',
+      '--target',
+      'helloPubSub',
+      '--signature-type',
+      'event',
+      '--port',
+      PORT,
+    ]);
+    const ffProcHandler = new Promise((resolve, reject) => {
+      let stdout = '';
+      let stderr = '';
+      ffProc.stdout.on('data', data => (stdout += data));
```

Comment:
Aside: @grant I'm curious what you'd think about adding a "shim" feature to the FFs.

(This would start other functions without requiring a separate process.)

If this sounds interesting to you, LMK and I'll look into drafting a proposal. 👍